src/swap/utils/sendSwapTransactions.ts

```diff
@@ -0,0 +1,31 @@
+import { Capabilities } from '../../constants';
+import type { SendSwapTransactionParams } from '../types';
+import { sendSingleTransactions } from './sendSingleTransactions';
+
+export async function sendSwapTransactions({
+  config,
+  sendCallsAsync,
+  sendTransactionAsync,
+  setCallsId,
+  updateLifecycleStatus,
+  walletCapabilities,
+  transactions,
+}: SendSwapTransactionParams) {
+  if (walletCapabilities[Capabilities.AtomicBatch]?.supported) {
+    // For batched transactions, we'll use `SwapProvider` to listen for calls to emit the `success` state
+    updateLifecycleStatus({
+      statusName: 'transactionPending',
+    });
+    const callsId = await sendCallsAsync({
+      calls: transactions,
+    });
```

Comment:
atomic batched transactions does not have a `transactionApproved` status since the approval transaction is part of the batch

Reply:
(discussed in API meeting)

Reply:
we need to modify the [swap messages](https://github.com/coinbase/onchainkit/blob/main/src/swap/utils/getSwapMessage.ts#L20-L26), currently transactionPending shows a confirm in wallet message, we only show Swap in progress on transactionApproved.        I think in this case, we want to show swap in progress without being in the approved status.  Could possibly add a walletApproved status that would also switch to the swap in progress message, thoughts?

Reply:
alternatively, we can have a third type for `transactionApproved`, but that's not how it's typically used today (`transactionApproved` is emitted after the transactions are confirmed on the blockchain)

```
statusName: 'transactionApproved',
statusData: {
   transactionType: 'batched'
}
```

Reply:
IMO we should get rid of the `Confirm in Wallet` state entirely, and the UI states should all just say `Transaction Pending` until the transaction has been confirmed, especially with the introduction of session keys

Reply:
https://github.com/coinbase/onchainkit/pull/1270/commits/3e8f7080a97972a0a58e587c96f480fc12fd9206

how does this look? now we use `transactionApproved` as a wallet approval in both cases

Reply:
and we are now using `transactionApproved` to mean a wallet approval